internal/ir/ir.go

```diff
@@ -352,6 +352,20 @@ type HashMap struct {
 	Value Value
 }
 
+// SubscriptExpr is an IR Value that represents the access of an element of an array.
+//
+// Example printed form:
+// %t0 = array["a","b","c"]
+// %t1 = "0"
+// %t2 = %t0[%t1]
+//
+// The SubscriptExpr implements the Value interface.
+type SubscriptExpr struct {
```

Comment:
I think that a better name for this should be `Lookup` since we will use this struct to lookup values from arrays, maps and any value that can be subscripted.